src/App.php

```diff
@@ -573,27 +561,17 @@ public function getUserAgent()
 			$this->getBaseURL();
 	}
 
-	/**
-	 * Checks if the site is called via a backend process
-	 *
-	 * @param Module $module The pre-loaded module (just name, not class!)
-
-	 * @return bool True, if the call is a backend call
-	 */
-	private function checkBackend(Module $module)
-	{
-		return basename(($_SERVER['PHP_SELF'] ?? ''), '.php') !== 'index' ||
-		       $module->isBackend();
-	}
-
 	/**
 	 * Returns true, if the call is from a backend node (f.e. from a worker)
 	 *
 	 * @return bool Is it a known backend?
+	 *
+	 * @deprecated 2019.09 - use App\Mode->isBackend() instead
+	 * @see App\Mode::isBackend()
 	 */
 	public function isBackend()
 	{
-		return $this->isBackend;
+		return $this->mode->isBackend();
```

Comment:
Is there any way to retrieve the Mode object outside of the App class?

Reply:
If not directly per parameter, at least per `BaseObject::getClass(App\Mode::class)` :-)

Reply:
btw. the last occurence of `App->isBackend()` is here:
https://github.com/friendica/friendica/blob/afac4b7698166cfbc2550f863aa792a2bb7beb7a/src/Core/Worker.php#L1127

Reply:
Got it, please add this to the deprecation notice for future reference.

Reply:
Yes, it's the one that crashed my node last time.

Reply:
> If not directly per parameter, at least per `BaseObject::getClass(App\Mode::class)` :-)

Got it, please add this to the deprecation notice.